File: D/1184-distance-between-bus-stops/main.go
package main

import "fmt"

func distanceBetweenBusStops(distance []int, start int, destination int) int {
	x, y := start, destination
	if start > destination {
		x, y = destination, start

	}
	sum, clock := 0, 0
	for i := 0; i < len(distance); i++ {
		sum += distance[i]
		if i >= x && i < y {
			clock += distance[i]
		}
	}
	if clock > sum-clock {
		return sum - clock
	}
	return clock
}

func main() {
	fmt.Println("a")
}

File: D/1184-distance-between-bus-stops/main.rs
struct Solution {}
impl Solution {
  pub fn distance_between_bus_stops(distance: Vec<i32>, start: i32, destination: i32) -> i32 {
    let sum = distance.iter().sum::<i32>();
    let s = start.min(destination) as usize;
    let d = start.max(destination) as usize;
    let v = (s..d).fold(0, |sum, idx| sum + distance[idx]);
    v.min(sum - v)
  }
}
